arm/Microsoft.KeyVault/vaults/deploy.bicep

```diff
@@ -148,9 +142,9 @@ var diagnosticsMetrics = [for metric in metricsToEnable: {
 }]
 
 var maxNameLength = 24
-var uniqueKeyVaultNameUntrim = uniqueString('Key Vault${baseTime}')
-var uniqueKeyVaultName = ((length(uniqueKeyVaultNameUntrim) > maxNameLength) ? substring(uniqueKeyVaultNameUntrim, 0, maxNameLength) : uniqueKeyVaultNameUntrim)
-var keyVaultName_var = (empty(keyVaultName) ? uniqueKeyVaultName : keyVaultName)
+var uniquenameUntrim = uniqueString('Key Vault${baseTime}')
```

Comment:
This will create a new key vault every time its run without a `name` parameter? Should we not link the name generation to something more specific to the key vault instance, i.e. subscription, rg and location?

Reply:
In theorie I'd agree, but I'd put that up for debate in a dedicated issue/meeting and not this PR as the logic was there before

Reply:
Good point!